src/Types/ArrayKey.php

```diff
@@ -13,20 +13,28 @@
 
 namespace phpDocumentor\Reflection\Types;
 
+use phpDocumentor\Reflection\PseudoType;
+use phpDocumentor\Reflection\Type;
+
 /**
  * Value Object representing a array-key Type.
  *
  * A array-key Type is the supertype (but not a union) of int and string.
  *
  * @psalm-immutable
  */
-final class ArrayKey extends AggregatedType
+final class ArrayKey extends AggregatedType implements PseudoType
 {
     public function __construct()
     {
         parent::__construct([new String_(), new Integer()], '|');
     }
 
+    public function underlyingType(): Type
+    {
+        return new String_();
```

Comment:
This seems incorrect to me, this type is not a string, it should return a `Compound` with the types `String_` and `Integer`. ArrayKey is an alternative notation for `String|Int`, not just `String`

Reply:
fixed